plugins/analytics-module-ga/src/apis/implementations/AnalyticsApi/GoogleAnalytics.ts

```diff
@@ -40,7 +45,7 @@ type CustomDimensionOrMetricConfig = {
  * Google Analytics API provider for the Backstage Analytics API.
  * @public
  */
-export class GoogleAnalytics implements AnalyticsApi {
+export class GoogleAnalytics implements AnalyticsApi, NewAnalyticsApi {
```

Comment:
Would it be worth it to introduce some kind of compatibility layer in `core-compat-api` instead of requiring all analytics api implementations to implement these types of changes?

e.g. a `CompatAnalyticsApi` interface that could be implemented = + a `convertLegacyAnalyticsEvent` function or something that handled the `extension`/`extensionId` logic?

Reply:
That's a good question, I had thought about it but wasn't sure if it is something we want to do... Great that you asked. What do you think @freben  and @Rugvip ? I would love to have a unique interface that deprecates context properties that we are going to drop support.

Reply:
I don't think it's really possible/valuable to do that tbh. The context data is very different in the two systems apart from the plugin ID. There's really no sane translation that I think is worth setting up between extension names in the old system and extension IDs in the new. Extension IDs are a lot more granular. Route ref IDs are gone in the new system too.